src/components/CanvasWrapper.tsx

```diff
@@ -125,40 +124,33 @@ export default function CanvasWrapper(inputProps: CanvasWrapperProps): ReactElem
   const props = { ...defaultProps, ...inputProps } as Required<CanvasWrapperProps>;
 
   // Access state properties
-  const backdropBrightness = useViewerStateStore((state) => state.backdropBrightness);
+  const currentFrame = useViewerStateStore((state) => state.currentFrame);
   const backdropKey = useViewerStateStore((state) => state.backdropKey);
-  const backdropSaturation = useViewerStateStore((state) => state.backdropSaturation);
   const backdropVisible = useViewerStateStore((state) => state.backdropVisible);
-  const categoricalPalette = useViewerStateStore((state) => state.categoricalPalette);
   const clearTrack = useViewerStateStore((state) => state.clearTrack);
   const collection = useViewerStateStore((state) => state.collection);
-  const colorRamp = useViewerStateStore((state) => state.colorRamp);
-  const colorRampRange = useViewerStateStore((state) => state.colorRampRange);
   const dataset = useViewerStateStore((state) => state.dataset);
   const datasetKey = useViewerStateStore((state) => state.datasetKey);
-  const featureKey = useViewerStateStore((state) => state.featureKey);
-  const inRangeLUT = useViewerStateStore((state) => state.inRangeLUT);
-  const objectOpacity = useViewerStateStore((state) => state.objectOpacity);
-  const outlierDrawSettings = useViewerStateStore((state) => state.outlierDrawSettings);
-  const outlineColor = useViewerStateStore((state) => state.outlineColor);
-  const outOfRangeDrawSettings = useViewerStateStore((state) => state.outOfRangeDrawSettings);
   const setBackdropVisible = useViewerStateStore((state) => state.setBackdropVisible);
   const setOpenTab = useViewerStateStore((state) => state.setOpenTab);
   const setTrack = useViewerStateStore((state) => state.setTrack);
   const showHeaderDuringExport = useViewerStateStore((state) => state.showHeaderDuringExport);
   const showLegendDuringExport = useViewerStateStore((state) => state.showLegendDuringExport);
   const showScaleBar = useViewerStateStore((state) => state.showScaleBar);
   const showTimestamp = useViewerStateStore((state) => state.showTimestamp);
-  const showTrackPath = useViewerStateStore((state) => state.showTrackPath);
-  const track = useViewerStateStore((state) => state.track);
-  const vectorConfig = useViewerStateStore(useShallow(selectVectorConfigFromState));
-  const vectorData = useViewerStateStore((state) => state.vectorMotionDeltas);
 
   const containerRef = useRef<HTMLDivElement>(null);
 
   const canv = props.canv;
   const canvasPlaceholderRef = useRef<HTMLDivElement>(null);
 
+  // Add subscriber so canvas parameters are updated when the state changes.
+  useEffect(() => {
+    return useViewerStateStore.subscribe(canvasStateParamsSelector, (params) => {
+      canv.setParams(params);
+    });
+  }, []);
+
```

Comment:
Subscriber means that the canvas can be updated outside of the React render lifecycle! CanvasWrapper doesn't need to rerender when properties change.